src/dfx/src/lib/canister_info.rs

```diff
@@ -202,7 +202,7 @@ impl CanisterInfo {
         &self.args
     }
 
-    pub fn get_shrink(&self) -> bool {
+    pub fn get_shrink(&self) -> Option<bool> {
         self.shrink
     }
```

Comment:
I'm having a hard time following `!matches!(canister_info.get_shrink(), Some(false))`.  How about this?
```suggestion
    pub fn get_shrink(&self) -> bool {
        match self.type_specific {
            CanisterTypeProperties::Rust { .. } | CanisterTypeProperties::Motoko => {
                self.shrink.unwrap_or(true)
            }
            CanisterTypeProperties::Custom { .. } | CanisterTypeProperties::Assets { .. } => {
                self.shrink.unwrap_or(false)
            }
        }
    }
```

Reply:
For future reference, `== Some(true)` works fine too.

Reply:
That `matches!` actually came from `cargo clippy`.

While moving such canister type specific logic into the general `CanisterInfo` is not good for encapsulation.
I would still keep the logic in each canister type builder and replace `matches` with more readable syntax.